### PR TITLE
Fix for ML-329 Tensor.At() on wrong indices

### PR DESCRIPTION
**Pull request by vhavrylov-ncube**
Fixed Tensor .At() accessor behavior when wrong index values are passed
Cleaned up existing unit tests
added tests for bad cases

### CODE REVIEW

File: libs/math/include/math/tensor.hpp [hunk 1]
```diff
@@ -459,13 +459,25 @@ class Tensor
   template <SizeType N, typename FirstIndex, typename... Indices>
   SizeType UnrollComputeColIndex(FirstIndex &&index, Indices &&... indices) const
   {
+    if (shape_[N] <= SizeType(index))
```
Comment (baykaner) [inline libs/math/include/math/tensor.hpp]: pedantic but shape_.at(N) is safer

Comment (vhavrylov-ncube) [reply libs/math/include/math/tensor.hpp]: sure!

File: libs/vm-modules/tests/unit/math/tensor_tests.cpp [hunk 2]
```diff
@@ -135,42 +111,68 @@ TEST_F(MathTensorTests, tensor_4_dim_fixed64_fill)
...
 
-TEST_F(MathTensorTests, tensor_set_on_invalid_index)
+TEST_F(MathTensorTests, tensor_at_invalid_index_count_too_few)
 {
   static char const *SRC = R"(
-    function main() : Tensor
+    function main()
       var tensor_shape = Array<UInt64>(1);
       tensor_shape[0] = 2u64;
+      tensor_shape[1] = 2u64;
```
Comment (baykaner) [inline libs/vm-modules/tests/unit/math/tensor_tests.cpp:154]: i don't think this really tests anything about tensor though, you're just setting up an Array with one element and then trying to access its second element. That fails before you even try to construct a tensor

**Review by baykaner — CHANGES_REQUESTED** [review]
looks like you have one irrelevant test (or at least a test that should live elsewhere), other than that it looks good

File: libs/vm-modules/tests/unit/math/tensor_tests.cpp (hunk 2 above)
Comment (vhavrylov-ncube) [reply libs/vm-modules/tests/unit/math/tensor_tests.cpp:154]: Oh, no, my fault here: Array<UInt64>(2) was intended; 2-dim tensor with 1-dim accessor called.

**Review by baykaner — APPROVED** [review]

**Pull request merged**